modules/ld/shacl-topquadrant-validator/src/main/scala/ch/epfl/bluebrain/nexus/commons/shacl/topquadrant/ValidationReport.scala

```diff
@@ -0,0 +1,60 @@
+package ch.epfl.bluebrain.nexus.commons.shacl.topquadrant
+
+import ch.epfl.bluebrain.nexus.commons.http.syntax.circe._
+import ch.epfl.bluebrain.nexus.commons.shacl.topquadrant.Vocabulary._
+import ch.epfl.bluebrain.nexus.commons.test.Resources.jsonContentOf
+import ch.epfl.bluebrain.nexus.rdf.Graph
+import ch.epfl.bluebrain.nexus.rdf.Graph._
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import ch.epfl.bluebrain.nexus.rdf.Vocabulary._
+import ch.epfl.bluebrain.nexus.rdf.syntax.circe._
+import ch.epfl.bluebrain.nexus.rdf.syntax.circe.context._
+import ch.epfl.bluebrain.nexus.rdf.syntax.jena._
+import ch.epfl.bluebrain.nexus.rdf.syntax.node.encoder._
+import ch.epfl.bluebrain.nexus.rdf.syntax.node.unsafe._
+import io.circe.Json
+import journal.Logger
+import org.apache.jena.rdf.model.Resource
+
+import scala.util.{Success, Try}
+
+/**
+  * Data type that represents the outcome of validating data against a shacl schema.
+  *
+  * @param conforms      true if the validation was successful and false otherwise
+  * @param json          the detailed message of the validator
+  */
+final case class ValidationReport(conforms: Boolean, targetedNodes: Int, json: Json) {
+
+  /**
+    * @return true if the validation report has been successful or false otherwise
+    */
+  def isValid: Boolean = targetedNodes > 0 && conforms
```

Comment:
we should probably have something like:

`def isValid(checkTargetCount: Boolean = true): Boolean` to have the option of disabling this behaviour.